File: translation_output/w_test/gpt-4/codenet/python/java/p03797.java
Here's how you could translate the provided Python code to Java:

```java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int s = scanner.nextInt();
        int c = scanner.nextInt();
        
        if (s * 2 <= c) {
            System.out.println(s + ((c - s * 2) / 4));
        } else {
            System.out.println(c / 2);
        }
    }
}
```

This Java code takes two integers as input and prints the output according to the provided logic. It first reads the input using a `Scanner` object, then checks the condition similar to the ternary operation in the python code and prints the result accordingly.

File: translation_output/w_test/gpt-4/codenet/java/python/p03797.py
Here is the Java code translated to Python:

```python
def find_sum(N, M):
    sum = 0
    if N * 2 > M:
        sum = M // 2
    elif M > N * 2:
        sum = N
        sum = sum + ((M - N * 2) // 4)
    return sum

# Test input:
N, M = map(int, input().split())

# Expected output:
print(find_sum(N, M))
```

When you provide the input "1 6", this code will output "2", just as expected.
